action.yml

```diff
@@ -11,6 +11,13 @@ inputs:
   input:
     description: 'The input path.'
     default: '.'
+    required: false
 runs:
-  using: 'node12'
-  main: './dist/main.js'
+  using: composite
+  steps:
+    - name: push
+      shell: bash
+      env:
+        BUF_TOKEN: ${{ inputs.buf_token }}
```

Comment:
If the user specifies `BUF_TOKEN` in the `env` themselves, does this overwrite it? For example, see the `buf-setup-action` docs [here](https://github.com/bufbuild/buf-setup-action#buf-token).

Reply:
I believe it does, but I'll do a test to double check that.

Reply:
Ok, well let's make sure we account for the `BUF_TOKEN` that could be set by the user themselves. If they set up their `env` likje the `buf-setup-action` docs suggests, we should make sure that we don't accidentally overwrite the `BUF_TOKEN` value with `inputs.buf_token` if `inputs.buf_token` is empty.

Reply:
`inputs.buf_token` is required, perhaps it's best to only expose one way of setting the token? It's easy enough to pass the environment variable into the input.

Reply:
We probably want to remove our guidance in `buf-setup-action` then so that we use `buf_token` parameters across the board. Otherwise it'll be too easy to mess this up.

Reply:
I double checked and this does indeed override the env set in the workflow.

`inputs.buf_token` is required, so I don't think there would be an expectation that setting a `BUF_TOKEN` environment variable would override it.